libr/anal/esil2reil.c

```diff
@@ -626,57 +699,72 @@ static int reil_poken(RAnalEsil *esil, ut8 n) {
 	RAnalReilInst *ins;
 	RAnalReilArg *op2, *op1;
 
-	op2 = reil_pop_arg(esil);
+	op2 = reil_pop_arg (esil);
 	if (!op2) return false;
-	op1 = reil_pop_arg(esil);
+	op1 = reil_pop_arg (esil);
 	if (!op1) {
 		R_FREE (op2);
 		return false;
 	}
 
 	if (op1->type != ARG_ESIL_INTERNAL) {
 		ins = R_NEW0 (RAnalReilInst);
+		if (!ins) return false;
 		ins->opcode = REIL_LDM;
 		ins->arg[0] = op2;
 		ins->arg[1] = R_NEW0(RAnalReilArg);
+		if (!ins->arg[1]) {
+			R_FREE (op1);
+			reil_free_inst (ins);
+			return false;
+		}
 		ins->arg[2] = R_NEW0(RAnalReilArg);
-		reil_make_arg(esil, ins->arg[1], " ");
-		get_next_temp_reg(esil, tmp_buf);
-		reil_make_arg(esil, ins->arg[2], tmp_buf);
+		if (!ins->arg[2]) {
+			R_FREE (op1);
+			reil_free_inst (ins);
+			return false;
+		}
+		reil_make_arg (esil, ins->arg[1], " ");
+		get_next_temp_reg (esil, tmp_buf);
+		reil_make_arg (esil, ins->arg[2], tmp_buf);
 		ins->arg[2]->size = ins->arg[0]->size;
-		reil_print_inst(esil, ins);
-		snprintf(esil->Reil->old, sizeof (esil->Reil->old) - 1, "%s:%d",
+		reil_print_inst (esil, ins);
+		snprintf (esil->Reil->old, sizeof (esil->Reil->old) - 1, "%s:%d",
 				ins->arg[2]->name, ins->arg[2]->size);
-		snprintf(esil->Reil->cur, sizeof (esil->Reil->cur) - 1, "%s:%d", op2->name,
+		snprintf (esil->Reil->cur, sizeof (esil->Reil->cur) - 1, "%s:%d", op2->name,
 				op2->size);
 		esil->lastsz = n * 8;
-		reil_push_arg(esil, op1);
-		reil_push_arg(esil, op2);
-		R_FREE(op1);
-		reil_free_inst(ins);
+		reil_push_arg (esil, op1);
+		reil_push_arg (esil, op2);
+		R_FREE (op1);
+		reil_free_inst (ins);
 	} else {
-		reil_flag_spew_inst(esil, op1->name + 1);
-		R_FREE(op1);
-		op1 = reil_pop_arg(esil);
-		reil_push_arg(esil, op2);
-		reil_push_arg(esil, op1);
-		R_FREE(op2);
-		R_FREE(op1);
+		reil_flag_spew_inst (esil, op1->name + 1);
+		R_FREE (op1);
+		op1 = reil_pop_arg (esil);
+		reil_push_arg (esil, op2);
+		reil_push_arg (esil, op1);
+		R_FREE (op2);
+		R_FREE (op1);
 	}
 
 	ins = R_NEW0 (RAnalReilInst);
+	if (!ins) return false;
 	ins->opcode = REIL_STM;
-	ins->arg[2] = reil_pop_arg(esil);
-	ins->arg[0] = reil_pop_arg(esil);
-	ins->arg[1] = R_NEW0(RAnalReilArg);
+	ins->arg[2] = reil_pop_arg (esil);
+	ins->arg[0] = reil_pop_arg (esil);
+	ins->arg[1] = R_NEW0 (RAnalReilArg);
+	if (!ins->arg[1]) {
+		reil_free_inst (ins);
```

Comment:
I think we need a `return` here. Otherwise on the next instruction ins will be dereferenced.